lib/matplotlib/backend_managers.py

```diff
@@ -74,6 +72,44 @@ def __init__(self, canvas):
         self.keypresslock = widgets.LockDraw()
         self.messagelock = widgets.LockDraw()
 
+        self._figure = None
+        if figure:
+            self.set_figure(figure)
+
+    @property
+    def canvas(self):
+        """Canvas managed by FigureManager"""
+        return self._figure.canvas
```

Comment:
This looks odd to me: if a default `ToolManager` is created and its first property, `canvas`, is accessed, it will raise AttributeError because `self._figure` will be None.  Presumably this won't happen ordinarily, but it just doesn't look right. Maybe `canvas` should trap it and raise a more informative exception.